src/parsers/time-parser.js

```diff
@@ -31,8 +28,13 @@ export default class TimeParser {
       const start = date.start ? Number(date.start.date()) : null;
       const end = date.end ? Number(date.end.date()) : null;
       const extractedText = date.text;
-      const processedText = text.substr(0, date.index) +
-        text.substr(date.index + extractedText.length);
+      let beforeText = text.substr(0, date.index).trim();
+      const afterText = text.substr(date.index + extractedText.length).trim();
+      if (/ (?:on|by)$/.test(beforeText)) {
```

Comment:
That looks really hacky for me. This should really be handled by a custom chrono parser.
If it's only a temporary fix, I'd add a comment.